target_chains/cosmwasm/contracts/pyth/src/contract.rs

```diff
@@ -156,24 +157,38 @@ pub fn execute(
     }
 }
 
-// TODO: add tests for these
 #[cfg(not(feature = "osmosis"))]
 fn is_fee_sufficient(deps: &Deps, info: MessageInfo, data: &[Binary]) -> StdResult<bool> {
     use cosmwasm_std::has_coins;
+
     let state = config_read(deps.storage).load()?;
-    return Ok(has_coins(
-        info.funds.as_ref(),
-        &get_update_fee(&deps, data)?,
-    ));
+
+    // for any chain other than injective there is only one base denom
+    // if base denom is present in coins and has enough amount this will return true
+    // or if the base fee is set to 0
+    // else it will return false
+    return Ok(state.fee.amount.u128() == 0
+        || has_coins(info.funds.as_ref(), &get_update_fee(deps, data)?));
 }
 
 // it only checks for fee denoms other than the base denom
 #[cfg(feature = "osmosis")]
-fn is_allowed_tx_fees_denom(deps: &Deps, denom: &String) -> StdResult<bool> {
+fn is_allowed_tx_fees_denom(deps: &Deps, denom: &String) -> bool {
+    // TxFeesQuerier uses stargate queries which we can't mock as of now.
+    // The capability has not been implemented in `cosmwasm-std` yet.
+    // Hence hacking it this way to be able to right tests.
+    // FIXME
```

Comment:
Use complete sentences in your comments. "Hence, we are hacking it with a feature flag ..."